packages/jsii-pacmak/lib/targets/go/types/go-type-reference.ts

```diff
@@ -74,66 +75,109 @@ export class GoTypeRef {
    * Return the name of a type for reference from the `Package` passed in
    */
   public scopedName(scope: Package): string {
-    return this.scopedTypeName(scope, false);
+    return this.scopedTypeName(this.typeMap, scope, false);
   }
 
   public scopedInterfaceName(scope: Package): string {
-    return this.scopedTypeName(scope, true);
+    return this.scopedTypeName(this.typeMap, scope, true);
   }
 
   public scopedReferenceName(scope: Package): string {
-    return this.scopedTypeName(scope, false, true);
+    return this.scopedTypeName(this.typeMap, scope, false, true);
   }
 
-  private scopedTypeName(
-    scope: Package,
-    asInterface: boolean,
-    asRef = false,
-  ): string {
-    if (this.primitiveType) {
-      return this.primitiveType;
+  public get typeMap(): TypeMap {
+    return this.buildTypeMap(this);
```

Comment:
That looks like something that perhaps should be stored (it's used in 3 distinct locations...).